module/os/freebsd/zfs/zfs_vnops_os.c

```diff
@@ -5772,6 +5775,17 @@ zfs_freebsd_pathconf(struct vop_pathconf_args *ap)
 	case _PC_HAS_HIDDENSYSTEM:
 		*ap->a_retval = 1;
 		return (0);
+#endif
+#ifdef _PC_CLONE_BLKSIZE
+	case _PC_CLONE_BLKSIZE:
+		zfsvfs = (zfsvfs_t *)ap->a_vp->v_mount->mnt_data;
+		if (zfs_bclone_enabled &&
+		    spa_feature_is_enabled(dmu_objset_spa(zfsvfs->z_os),
+		    SPA_FEATURE_BLOCK_CLONING))
+			*ap->a_retval = zfs_max_recordsize;
```

Comment:
I can't say that I like `zfs_max_recordsize`, since it does not represent the pool, but instead more of a system capabilities.  In most cases it will be 16MB, which is rarely reached. Same time on i386 pools imported from another system might have blocks bigger than its `zfs_max_recordsize`.  I think you could say whether dataset ever had any block above 128K by checking `dsl_dataset_feature_is_active(ds, SPA_FEATURE_LARGE_BLOCKS)`.  But otherwise I'd say that recordsize property is the best guess we have.